google-http-client-apache-v2/src/test/java/com/google/api/client/http/apache/v2/ApacheHttpTransportTest.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicHttpResponse;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.protocol.HttpRequestExecutor;
+import org.junit.Test;
+
+/**
+ * Tests {@link ApacheHttpTransport}.
+ *
+ * @author Yaniv Inbar
+ */
+public class ApacheHttpTransportTest {
+
+  @Test
+  public void testApacheHttpTransport() {
+    ApacheHttpTransport transport = new ApacheHttpTransport();
+    checkHttpTransport(transport);
+  }
+
+  @Test
+  public void testApacheHttpTransportWithParam() {
+    ApacheHttpTransport transport = new ApacheHttpTransport(HttpClients.custom().build());
+    checkHttpTransport(transport);
+  }
+
+  @Test
+  public void testNewDefaultHttpClient() {
+    HttpClient client = ApacheHttpTransport.newDefaultHttpClient();
+    checkHttpClient(client);
+  }
+
+  private void checkHttpTransport(ApacheHttpTransport transport) {
+    assertNotNull(transport);
+    HttpClient client = transport.getHttpClient();
+    checkHttpClient(client);
+  }
+
+  private void checkHttpClient(HttpClient client) {
+    assertNotNull(client);
+    // TODO(chingor): Is it possible to test this effectively? The newer HttpClient implementations
+    // are read-only and we're testing that we built the client with the right configuration
+  }
+
+  @Test
+  public void testRequestsWithContent() throws Exception {
+    HttpClient mockClient = mock(HttpClient.class);
+    HttpResponse mockResponse = mock(HttpResponse.class);
+    when(mockClient.execute(any(HttpUriRequest.class))).thenReturn(mockResponse);
+
+    ApacheHttpTransport transport = new ApacheHttpTransport(mockClient);
+
+    // Test GET.
+    subtestUnsupportedRequestsWithContent(
+        transport.buildRequest("GET", "http://www.test.url"), "GET");
+    // Test DELETE.
+    subtestUnsupportedRequestsWithContent(
+        transport.buildRequest("DELETE", "http://www.test.url"), "DELETE");
+    // Test HEAD.
+    subtestUnsupportedRequestsWithContent(
+        transport.buildRequest("HEAD", "http://www.test.url"), "HEAD");
+
+    // Test PATCH.
+    execute(transport.buildRequest("PATCH", "http://www.test.url"));
+    // Test PUT.
+    execute(transport.buildRequest("PUT", "http://www.test.url"));
+    // Test POST.
+    execute(transport.buildRequest("POST", "http://www.test.url"));
+    // Test PATCH.
+    execute(transport.buildRequest("PATCH", "http://www.test.url"));
+  }
+
+  private void subtestUnsupportedRequestsWithContent(ApacheHttpRequest request, String method)
+      throws Exception {
+    try {
+      execute(request);
+      fail("expected " + IllegalArgumentException.class);
+    } catch (IllegalArgumentException e) {
+      // expected
+      assertEquals(e.getMessage(),
+          "Apache HTTP client does not support " + method + " requests with content.");
+    }
+  }
+
+  private void execute(ApacheHttpRequest request) throws Exception {
+    byte[] bytes = StringUtils.getBytesUtf8("abc");
```

Comment:
Outside the scope of this PR, but StringUtils was a bad idea. This line is trivial to implement without an extra dependency.